qemu/tests/virtio_trace_pipenb.py

```diff
@@ -0,0 +1,52 @@
+import logging
+from avocado.utils import process
+from virttest import error_context
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Under named-pipe non-blocking testing:
+    1) Make FIFO per CPU in a host.
+    2) reboot guest with virtio-serial device, control path and data path per CPU.
+    3) Download trace agent and compile it.
+    4) Enable ftrace in the guest.
+    5) Run trace agent in the guest.
+    6) Open FIFO in a host.
+    7) Start to read trace data by ordering from a host
+    8) Stop to read trace data by ordering from a host
+    9) repeat 7) and 8) with different CPU
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+
+    def execute_host_cmd(host_cmd, timeout=60):
+        """
+        Execute the host_cmd on host, limited in timeout period
+
+        param host_cmd: the host_cmd to run on host
+        param timeout: the timeout for running this command
+        return: the output of the host_cmd
+        """
+        logging.info("Executing host command: %s" % host_cmd)
+        cmd_result = process.getstatusoutput(host_cmd, timeout=timeout, shell=True)
+        return cmd_result
+
+    timeout = float(params.get("login_timeout", 240))
+    v_path = "/tmp/serial0"
+    execute_host_cmd("mkfifo {}".format(v_path))
+    extra_params = params.get("extra_params", "")
+    extra_params += " -chardev pipe,id=ch0,path=/tmp/serial0 "
+    extra_params += " -device virtserialport,chardev=ch0,name=serial0 "
```

Comment:
you can support creating pipe path like this:
https://github.com/avocado-framework/avocado-vt/blob/master/virttest/qemu_devices/qcontainer.py#L2116

        ```
if backend in ['unix_socket', 'file', 'serial',
                       'tty', 'parallel', 'parport']:
            chardev_param.update({'path': file_name})
        elif backend in ['pipe']:
            process.system("mkfifo %s" % file_name)
            chardev_param.update({'path': file_name})
```

Reply:
Hi there is one problem, it does not set the name for virtserialport, and for the test it need to set the name

Reply:
Also the auto generated path for some reason does not create anything on host

-chardev pipe,path=/var/tmp/avocado_1jvqcxlu/serial-vs1-20200110-035705-uwhvnRjT,id=chardev_vs1: Could not open '/var/tmp/avocado_1jvqcxlu/serial-vs1-20200110-035705-uwhvnRjT': No such file or directory"

Reply:
> Also the auto generated path for some reason does not create anything on host
> 
> -chardev pipe,path=/var/tmp/avocado_1jvqcxlu/serial-vs1-20200110-035705-uwhvnRjT,id=chardev_vs1: Could not open '/var/tmp/avocado_1jvqcxlu/serial-vs1-20200110-035705-uwhvnRjT': No such file or directory"

You need to re-install avocado-vt to enforce the code changes.